MapleServer2/Tools/TaskScheduler.cs

```diff
@@ -9,13 +9,13 @@ public class TaskScheduler
 
     public static TaskScheduler Instance => _instance ??= new();
 
-    public void ScheduleTask(int hour, int min, double intervalInHour, Action task)
+    public void ScheduleTask(int hour, int min, double intervalInHour, int minutesToRunPerDay, Action task)
     {
         DateTime now = DateTime.Now;
         DateTime firstRun = new(now.Year, now.Month, now.Day, hour, min, 0, 0);
-        if (now > firstRun)
+        while (now > firstRun)
         {
-            firstRun = firstRun.AddDays(1);
+            firstRun = firstRun.AddMinutes(minutesToRunPerDay);
```

Comment:
I'm not sure I follow the logic here, I think this was fine before?
We schedule an event for a certain time during the day.
If that time has already passed today, we add 1 day on to the event.
We then figure out how long it's going to be until we run it next, and tell the timer to run for the first time at that time, and then at the interval after.

Reply:
These events run multiple times per day. If I kept it as days, the first time the server starts, the events would never trigger on the first day.